main_test.go

```diff
@@ -58,6 +58,14 @@ func TestCompiler(t *testing.T) {
 	if runtime.GOOS != "windows" {
 		t.Run("Host", func(t *testing.T) {
 			runPlatTests("", matches, t)
+			if runtime.GOOS == "darwin" {
+				for _, env := range [][2]string{{"ENV1", "VALUE1"}, {"ENV2", "VALUE2"}} {
+					if err := os.Setenv(env[0], env[1]); err != nil {
+						t.Fatal(err)
+					}
+				}
+				runTest("testdata/libc/env.go", "", t)
```

Comment:
This modifies global state, which I think is a bad idea. Perhaps you can change the second parameter of `runtTest` to accept a slice of environment variables?

```suggestion
				runTest("testdata/libc/env.go", "", t, []string{"ENV1=VALUE1", "ENV2=VALUE2"})
```